proposals/0298-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -145,80 +151,18 @@ If a parameter includes an unknown enum value, SDL Core has to cut off such valu
-      * Application HMI type(s) (`AppHMIType` array)
-
-#### Changes related to sending request(s) from mobile application to SDL Core
-  After the connection is established and mobile application is registered with SDL Core, mobile application works as usual but requires the request parameters validation. An issue will occur when the mobile application calls a method and passes an enum values array as a parameter that belongs to the higher SDL version. In this case, the Mobile Proxy should filter out unsupported params before sending the request and provide warning in logs as `unsupported_values`.
-
-  The validation sequence:
-
-  1. Mobile application sends request to SDL Core through the proxy.
-  2. The proxy checks the request parameters:
-     * If there are no enum values unknown to SDL, the proxy forwards the request to SDL Core without changes.
-
-     * If some of the enum values are unknown to SDL Core the proxy should:
-          * Filter out all RPCs' unsupported enum values including nested RPCs if those are present, before making a request to make sure it is compliant with the current spec
-          * Log a warning level with «unsupported_values» (enum’s/struct’s names)
-
-  **Note**: For `RegisterAppInterface` to filter we should use version of PROTOCOL, but not the MOBILE_API according to system’s flow. **iOS** should be extended with similar mechanism in order to encapsulate filtering logic.
-
-  **For example** :
-  If the UI app requested unsupported `AppHMIType` and version which are not compatible, then the proxy must filter unsupported HMI types.
-
-   - Before filtering :
-       ```
-       "AppHMIType" : ["DEFAULT", "MEDIA", "UKNOWN"]
-       ```
-
-   - After filtering :
-       ```
-       AppHMIType" : ["DEFAULT", "MEDIA"]
-       ```
-
-  ***Note**: In case no HMI types are left after filtering, Mobile Library should throw an exception, otherwise `RegisterAppInterface` RPC will be sent with remaining HMI types.*
-
-  ***Note**: Now system flow is the following: if `RegisterAppInterface` fails with an error then Mobile library should pass the error status to the Mobile Library adding failure reason.*
-
-  ***Note**: The protocol version control logic is partly implemented in Android and not implemented in iOS. However, what is implemented has its flaws and should be updated following the logic above.*
-
-
 ## Potential downsides
-  In the case, new RPCs would added, the new structures/enums should be implemented with additional implementation of filtering mechanism for particular objects.
 
-  **Note:** For `RegisterAppInterface` we cannot keep connection if invalid param is present due to the current flow of the older SDL versions. The connection can be dropped (End Session) by SDL Core but not by mobile device (SDL library).
+There is the case when a Mobile Application has the older API version (for example 6.0) then SDL Core and the SDL Core has the newer API version (for example 7.0).
```

Comment:
```suggestion
There is the case when a Mobile Application has an older Mobile API version (for example 6.0) than SDL Core and SDL Core has the newer Mobile API version (for example 7.0).
```

Reply:
Updated in 3e4835a64afd1b1977009b77d4907a68f37120ee